magefile.go

```diff
@@ -8,31 +8,153 @@
 package main
 
 import (
+	"context"
+	"fmt"
+	"os"
 	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
 
-	//mage:import
 	"github.com/elastic/elastic-agent-libs/dev-tools/mage"
+	devtools "github.com/elastic/elastic-agent-shipper/dev-tools/common"
+	"github.com/elastic/elastic-agent-shipper/tools"
+
+	//mage:import
+
 	"github.com/elastic/elastic-agent-libs/dev-tools/mage/gotool"
 	"github.com/magefile/mage/mg"
 	"github.com/magefile/mage/sh"
 )
 
-var (
-	goLicenserRepo = "github.com/elastic/go-licenser@v0.4.1"
+const (
+	GoreleaserRepo = "github.com/goreleaser/goreleaser@v1.6.3"
 )
 
 // Aliases are shortcuts to long target names.
 // nolint: deadcode // it's used by `mage`.
 var Aliases = map[string]interface{}{
-	"llc":  mage.Linter.LastChange,
-	"lint": mage.Linter.All,
+	"build":    Build.Binary,
+	"unitTest": Test.Unit,
+}
+
+// BUILD
+
+// Build contains targets related to linting the Go code
+type Build mg.Namespace
+
+// All builds binaries for the all  os/arch.
+func (Build) All() {
+	os.Setenv("PLATFORM", "all")
+	mg.Deps(Build.Binary)
+}
+
+// Clean removes the build directory.
+func (Build) Clean(test string) {
+	os.RemoveAll("build") // nolint:errcheck //not required
+}
+
+// TestBinaries checks if the binaries are generated (for now).
+func (Build) TestBinaries() error {
```

Comment:
Could this be renamed to `CheckBinaries` so that it is clear it doesn't actually run any tests (go test)? That seems to be the convention for other mage targets.